app/controllers/products_controller.rb

```diff
@@ -0,0 +1,51 @@
+class ProductsController < ApplicationController
+
+  def index
+    @categories = Category.order(name: :asc)
+    @merchants = User.joins(:products).uniq.sort
+
+    scope = Product.where(retired: false).order(name: :asc)
+
+    if params[:category].present?
+      scope = scope.joins(:categories).where(categories: {id: params[:category]})
+    end
+
+    if params[:merchant].present?
+      scope = scope.where(user_id: params[:merchant])
+    end
+
+    @products = scope
```

Comment:
Okay, now that I've made it all the way through I can say that your method is doing a lot!

You have done a great job at keeping it as terse as possible and have used a lot of fancy active_record methods to get there. 

Betsy provides a lot of really intense learning opportunities. One of them is teamwork. If a bug arose in this code, would your team-mates be able to dissect this code enough to find and fix the bug? If so, that's awesome and you have one of the more successful Betsy teams I've ever seen. If not, this might be a good time to try to make the code more readable. 

Options that might make this more readable:
* Named scopes in the Products model so that at each stage teammates have a name for the collection of products they are working with.
* Comments. Usually in production code in Ruby there aren't a lot of comments. But the more complicated the endeavor, especially with things like joins, the more valid they would be. And especially in an educational context it can really help your teammates to more quickly get up to speed on what your code is doing.
* Separate methods. This probably isn't the way that I would go for this particular method but even just having methods that are well named can help your teammates follow the thread from one step to the next because the return value of the smaller method could have a clear label for the collection they are now holding in their heads.